### PR TITLE
Add instruction to access DC2 catalogs at NERSC to CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by combet**
Title says it all.
- [x] Check notebooks from issue #249 work for others, when the updated instructions from this issue (#250) are followed.

### CODE REVIEW

**Comment by cavestruz** [comment]
Hi @combet -  thank you for this PR!  

**Installing: ** @pennalima and I just went through the documentation for CONTRIBUTING.md (start-to-finish for @pennalima and the last addition of the GCR catalogs for me).  The installs worked for both of us on NERSC (yay!).  There are minor additions we committed (line to specify that a user should run those line-by-line, as it seems that depending on the command, it doesn't necessarily work to copy and paste a whole block for @pennalima ).  Also, there is a minor addition reminding the user that the source activate line needs to happen at the start to a user's session in nersc, meaning they need to add `module load python` either to their .bashrc or the start to their workflow, and to make sure they are in the correct conda environment at the start to each workflow.  

**Testing: ** We wanted to test the modified notebooks in Issue #249 with the updated clmmenv conda environment.  We are able to:
`import GCRCatalogs`
but, we cannot load the catalog (see below).

This should likely be linked to the PR for Issue #249 as well, but:
- [ ] In data_and_model_demo_DC2.ipynb: `extragalactic_cat = GCRCatalogs.load_catalog('cosmoDC2_v1.1.4_small')`
results in a: 

> `/.conda/envs/clmmenv/lib/python3.8/site-packages/GCRCatalogs/cosmodc2.py in <module>
>      11 import yaml
>      12 import numpy as np
> ---> 13 import h5py
>      14 import healpy as hp
>      15 from astropy.cosmology import FlatLambdaCDM
> 
> ModuleNotFoundError: No module named 'h5py'`.
> 
**This is possibly an issue for the GCRCatalogs repo, but wasn't sure if you'd separately installed h5py - if it is a separate issue for the GCRCatalogs repo, we can create that over there.

- [ ] In DC2_gt_profiles.ipynb: `import FoFCatalogMatching` results in a:

>  `ModuleNotFoundError: No module named 'FoFCatalogMatching'`

Question - was FoFCatalogMatching something we should have access to from the `pip install https://github.com/LSSTDESC/gcr-catalogs/archive/master.zip
`?  Or does this point to something that needs to also be added to CONTRIBUTING.md?  Also a question for @marina-ricci , since it seems you were able to get these notebooks working?  Might be a CC vs. NERSC specific thing.

Thanks!  
- Camille and @pennalima 

P.S.  @pennalima is still adding some in-line review comments that also incorporate @vitenti 's notes from when he went through the installation on NERSC.

**Comment by combet** [comment]
@cavestruz, @pennalima, thanks for testing this out!
- `FoFCatalogMatching` is required for the `gt_profiles` notebook that I added after having written these instructions. It comes with the DESC conda environment, but needs to be installed by the user using `pip install https://github.com/yymao/FoFCatalogMatching/archive/master.zip`if building their own environment. I've updated the instructions.
- I do not recall installing `h5py`, but I see it in my `clmmenv` environment, so must have at some point. Sorry I overlooked that. It also needs to be install using `pip install h5py`. Instructions updated.

**Comment by pennalima** [comment]
Hi @combet  and @cavestruz , 
I tested it again following your updates. 
The example did not run because there is no "ShearAzimuthalAverager". I looked for it and I think this is an old module. 
So I commented the line "from clmm import ShearAzimuthalAverager" in the "data_and_model..." notebook to see what I would get.

The next error happens when loading the catalog (GCRCatalogs.load_catalog('cosmoDC2_v1.1.4_small')) .
 12 import numpy as np
     13 import h5py
---> 14 import healpy as hp
     15 from astropy.cosmology import FlatLambdaCDM
     16 from GCR import BaseGenericCatalog

ModuleNotFoundError: No module named 'healpy'

So I installed healpy and the example ran until the function ShearAzimuthalAverager was called: 
NameError                                 Traceback (most recent call last)
<ipython-input-6-0091b3ae28d8> in <module>
      1 # create an object, given cluster dictionary and galaxy astropy table
----> 2 saa = ShearAzimuthalAverager(cl_dict, dat)
      3 
      4 # compute tangential and cross shear for each galaxy
      5 saa.compute_shear()

NameError: name 'ShearAzimuthalAverager' is not defined 

The same happens for the "shear_azimuthal..." notebook.

In short, we just need to include *pip install healpy* in the instructions in order to load, read and manipulate the catalog. 
To go further  we have to update the examples importing the "good" module (polaraveraging?).

**Comment by combet** [comment]
@pennalima, sorry if this wasn't clear: the up-to-date DC2 notebooks are in an other branch/issue. Issue 250 was just to **update the instructions** in CONTRIBUTING for NERSC access. 
So please, try the notebooks from branch Issue/249/dc2 notebooks.

**Comment by pennalima** [comment]
Thanks @combet . I just addressed the right issue. :)

**Comment by combet** [comment]
I've updated CONTRIBUTING following the missing imports @pennalima identified when running the notebooks from issue #249. I think all is in order now. Flagging this ready for review.

**Comment by pennalima** [comment]
Hi @combet, the conflicts must be solved before. It seems simple to do, but I don't know which version should be kept.

**Comment by combet** [comment]
@pennalima, conflicts are now fixed. Please approve at your earliest convenience.

**Comment by pennalima** [comment]
> @pennalima, conflicts are now fixed. Please approve at your earliest convenience.
Done! @combet
